Blaze PersistenceUnit fails to start on Wildfly 27

<!--- This template is for bugs. Remove it for other issues. -->

<!--- Choose an expressive title -->
### Description
<!--- Explain what you did and maybe show some code excerpts -->
While updating an application to Wildfly 27 I encountered an error when the persistence unit is started. To ensure this has nothing to do with my project I set up a new repository and followed the Getting Started guide. The error is there too.

### Expected behavior
<!--- What outcome would you expect? -->
PersistenceUnit initializes and the application starts.

### Actual behavior
<!--- What happened that you didn't expect? -->
The following exception is thrown when deploying the application
```
[org.jboss.msc.service.fail] (ServerService Thread Pool -- 22) MSC000001: Failed to start service jboss.persistenceunit."blaze-persistence-demo-1.0-SNAPSHOT#ExampleDS": org.jboss.msc.service.StartException in service jboss.persistenceunit."blaze-persistence-demo-1.0-SNAPSHOT#ExampleDS": java.lang.NoClassDefFoundError: org/hibernate/annotations/common/reflection/ReflectionManager
	at org.jboss.as.jpa@27.0.0.Final//org.jboss.as.jpa.service.PersistenceUnitServiceImpl$1$1.run(PersistenceUnitServiceImpl.java:199)
	at org.jboss.as.jpa@27.0.0.Final//org.jboss.as.jpa.service.PersistenceUnitServiceImpl$1$1.run(PersistenceUnitServiceImpl.java:129)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.wildfly.security.elytron-base@2.0.0.Final//org.wildfly.security.manager.WildFlySecurityManager.doChecked(WildFlySecurityManager.java:664)
	at org.jboss.as.jpa@27.0.0.Final//org.jboss.as.jpa.service.PersistenceUnitServiceImpl$1.run(PersistenceUnitServiceImpl.java:214)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.JBossThread.run(JBossThread.java:513)
Caused by: java.lang.NoClassDefFoundError: org/hibernate/annotations/common/reflection/ReflectionManager
	at deployment.blaze-persistence-demo-1.0-SNAPSHOT.war//com.blazebit.persistence.integration.hibernate.Hibernate60MetadataContributor.addEntity(Hibernate60MetadataContributor.java:59)
	at deployment.blaze-persistence-demo-1.0-SNAPSHOT.war//com.blazebit.persistence.integration.hibernate.Hibernate60MetadataContributor.contribute(Hibernate60MetadataContributor.java:53)
	at org.hibernate@6.1.5.Final//org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:297)
	at org.hibernate@6.1.5.Final//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:1350)
	at org.hibernate@6.1.5.Final//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1421)
	at org.hibernate.jipijapa-hibernate6@27.0.0.Final//org.jboss.as.jpa.hibernate.TwoPhaseBootstrapImpl.build(TwoPhaseBootstrapImpl.java:45)
	at org.jboss.as.jpa@27.0.0.Final//org.jboss.as.jpa.service.PersistenceUnitServiceImpl$1$1.run(PersistenceUnitServiceImpl.java:171)
	... 10 more
Caused by: java.lang.ClassNotFoundException: org.hibernate.annotations.common.reflection.ReflectionManager from [Module "deployment.blaze-persistence-demo-1.0-SNAPSHOT.war" from Service Module Loader]
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:200)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	... 17 more
```

### Steps to reproduce
<!--- Give us enough details so we can create a testcase that reproduces this problem -->
<!--- Idealy you would create a test case based on the template project from: https://github.com/Blazebit/blaze-persistence-test-case-template -->
<!--- Either attach a ZIP containing the test case or create a pull request against the test-case-template repository and put the link to the PR here -->
1. Download and extract Wildfly 27 (https://www.wildfly.org/downloads/)
2. Deploy the code from https://github.com/theobisproject/blaze-persistence-demo



### Environment
<!--- Environment info like e.g.					-->
<!--- Version:            1.2.0-Alpha1				-->
<!--- JPA-Provider:       Hibernate 5.2.7.Final		-->
<!--- DBMS:               PostgreSQL 9.6.2			-->
<!--- Application Server: Java SE					-->

Version:            1.6.7
JPA-Provider:       Hibernate 6.1.5.Final
DBMS:               H2
Application Server: Wildfly 27

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,6 +36,7 @@ Changes that happened in releases
 * Always treat flat views for embeddable types as updates to support partial flushing
 * Fix QUERY flushing issues with MappingSingular attributes
 * Fix support for entity functions in DML statements
+* Fix deployment problems with Hibernate 6 integration on Wildfly 27
 
 ### Backwards-incompatible changes
 

--- a/integration/hibernate-6.0/pom.xml
+++ b/integration/hibernate-6.0/pom.xml
@@ -147,6 +147,27 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <artifactId>maven-jar-plugin</artifactId>
+                <inherited>true</inherited>
+                <configuration>
+                    <archive>
+                        <manifestEntries>
+                            <!-- Newer Wildfly versions don't export the commons annotations module anymore, so we need an explicit dependency -->
+                            <Dependencies>org.hibernate.commons-annotations</Dependencies>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>default-jar</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
         </plugins>
     </build>
 </project>

